### PR TITLE
fix(gopls) resolve top-level go.mod as root_dir

### PR DESCRIPTION
**Pull request by indeedhat**
This is just a fix for a bug previously reverted commit by @mjlbach  i didn't do any of the actual work

### CODE REVIEW

File: lua/lspconfig/gopls.lua [hunk 1]
```diff
@@ -6,7 +6,32 @@ configs.gopls = {
...
-      return util.root_pattern 'go.work'(fname) or util.root_pattern('go.mod', '.git')(fname)
+      -- First, search for go.work
+      local primary_root = util.root_pattern 'go.work'(fname)
+      if primary_root then
+        return primary_root
+      end
+
+      -- Then, search up the filesystem for go.mod direcrory
+      local go_mod_hierarchy = {}
+      if primary_root then
```
Comment (mjlbach) [inline lua/lspconfig/gopls.lua:17]: duplicate of line 11-13

Comment (indeedhat) [reply lua/lspconfig/gopls.lua:17]: i actually did notice that but as it was in the original commit i was working from i just left it in, ill remove it and do another commit

**Comment by glepnir** [comment]
this pr had conflict and Ci failed. not allow to use `dirname` function in `root_dir` in ci rules.

**Comment by indeedhat** [comment]
this was fixed by another commit quite a long while back, it can just be closed

**Pull request closed without merging**